src/app/js/src/App.jsx

```diff
@@ -32,23 +42,72 @@ class App extends Component {
         this.layers = [
             baseLayer,
         ];
+        this.layersCounter = 1;
         this.clearLayers = this.clearLayers.bind(this);
         this.share = this.share.bind(this);
         this.handleShareSbRequestClose = this.handleShareSbRequestClose.bind(this);
         this.handleErrorSbRequestClose = this.handleErrorSbRequestClose.bind(this);
         this.openAddLayerDialog = this.openAddLayerDialog.bind(this);
         this.addLayer = this.addLayer.bind(this);
         this.removeLayers = this.removeLayers.bind(this);
+        this.removeLayer = this.removeLayer.bind(this);
+        this.editLayer = this.editLayer.bind(this);
     }
 
     componentDidMount() {
         map.setTarget('map');
     }
 
-    addLayer(layer) {
-        map.addLayer(layer);
-        this.layers.push(layer);
-        return this.layers.length;
+    addLayer() {
+        if (this.props.url === '') {
+            return;
+        }
+        const newLayer = new TileLayer({
+            source: new XYZ({
+                url: this.props.url,
+            }),
+        });
+        let layerName = this.props.name;
+        if (layerName === '') {
+            layerName = `Layer ${this.layersCounter}`;
+            this.layersCounter += 1;
+        }
+        const newTileJSON = {
+            tilejson: '2.2.0',
+            name: layerName,
+            version: '1.0.0',
+            scheme: 'xyz',
+            tiles: [
+                this.props.url,
+            ],
+        };
+        const tileJSONList = this.props.tileJSON;
+        tileJSONList.push(newTileJSON);
+        this.props.dispatch(addLayer({
+            newLayer: {
+                name: layerName,
+                url: this.props.url,
+                tileJSON: newTileJSON,
+            },
+        }));
+        this.props.dispatch(changeTileJson({
+            tileJSON: tileJSONList,
+        }));
+
+        // Add layer to the ol map
+        map.addLayer(newLayer);
+        // Add layer to list of layers maintained
+        this.layers.push(newLayer);
+        // Clear artifacts of add
+        this.props.dispatch(toggleAddLayerDialog({
+            showAddLayerDialog: false,
+        }));
+        this.props.dispatch(changeLayerName({
+            name: '',
+        }));
+        this.props.dispatch(changeLayerUrl({
+            url: '',
+        }));
```

Comment:
Consider refactoring and reducing the length of this function. For example, the following could be pulled out into a separate "clear" function.

```
 this.props.dispatch(toggleAddLayerDialog({
 +            showAddLayerDialog: false,
 +        }));
 +        this.props.dispatch(changeLayerName({
 +            name: '',
 +        }));
 +        this.props.dispatch(changeLayerUrl({
 +            url: '',
 +        }));
```